selectel/dbaas.go

```diff
@@ -420,13 +427,34 @@ func resizeDatastore(ctx context.Context, d *schema.ResourceData, client *dbaas.
 	return nil
 }
 
-func validateDatastoreType(ctx context.Context, expectedDatastoreTypeEngine string, typeID string, client *dbaas.API) diag.Diagnostics {
+func containDatastoreType(expectedTypes []string, datastoreType string) bool {
+	for _, expectedType := range expectedTypes {
+		if expectedType == datastoreType {
+			return true
+		}
+	}
+
+	return false
+}
+
+func buildDatastoreTypeErrorMessage(expectedDatastoreTypeEngines []string, datastoreTypeEngine string) string {
+	var baseMessage string
+	if len(expectedDatastoreTypeEngines) > 1 {
+		baseMessage = "Provided datastore type must have one of the following engine types: "
+	} else {
+		baseMessage = "Provided datastore type must have an engine "
+	}
+
+	return baseMessage + strings.Join(expectedDatastoreTypeEngines, ", ") + " for this resource. But provided type is " + datastoreTypeEngine
+}
+
+func validateDatastoreType(ctx context.Context, expectedDatastoreTypeEngines []string, typeID string, client *dbaas.API) diag.Diagnostics {
 	datastoreType, err := client.DatastoreType(ctx, typeID)
 	if err != nil {
 		return diag.FromErr(errors.New("Couldnt get datastore type with id" + typeID))
```

Comment:
Maybe we should add space between message and `typeID`

Reply:
Fixed